### DISCUSSION TITLE
problem creating minimap after creating WaveSurfer

### DISCUSSION THREAD

**Discussion by Kalinkindev** [katspaugh/wavesurfer.js#3407] [Q&A]
help me please
I'm creating an audio playlist html, js

I have several playlists
each has audio and they can be similar

if audio is playing, active audio appears below
I need to create a minimap there when the audio plays

if I create a minimap for all audio at once, then I create a container
```
  var wavesurfer = WaveSurfer.create({
        container: webform,
        waveColor: "#bebebe",
        progressColor: "#000000",

        normalize: true,
        responsive: true,
        fillParent: true,
        height: 20,
        plugins: [
          Minimap.create({
            height: 30,
            container: playerActiveWafesurfer,
            waveColor: "#bebebe",
            progressColor: "#000000",
          }),
        ],
        url: `${audio.dataset.audio}`,
      });
```
object console.log(wavesurfer)
https://i.gyazo.com/cdd766218fe1b244c3a17f0b96607b95.png
if I create a minimap when the audio is playing, then I have "container: null" in the plugin object, but in the container option it is correct
```
      var wavesurfer = WaveSurfer.create({
        container: webform,
        waveColor: "#bebebe",
        progressColor: "#000000",

        normalize: true,
        responsive: true,
        fillParent: true,
        height: 20,
       
        url: `${audio.dataset.audio}`,
      });

 wavesurfer.plugins.pop();
        const minimap = Minimap.create({
          height: 30,
          container: playerActiveWafesurfer,
          waveColor: "#bebebe",
          progressColor: "#000000",
        });
        wavesurfer.plugins.push(minimap);
        
            
```
object console.log(wavesurfer)
https://i.gyazo.com/50937da75b2e8629515d181ee53512ff.png

**Comment by katspaugh** [comment]
You might want to take a look at this demo: https://wavesurfer.xyz/examples/?react-global-player.js

Instead of a Minimap, it creates a full-blown copy of a currently playing wavesurfer.

**Reply by Kalinkindev** [reply]
thanks, but this is in react((
I can’t rewrite it in js for my task from react
All I have left to do is make a minimap

**Reply by katspaugh** [reply]
In that case, you should replace

```js
wavesurfer.plugins.pop();
wavesurfer.plugins.push(minimap);
```

with

```js
const previousMinimap = wavesurfer.getActivePlugins().find((plugin) => plugin instanceof Minimap)
previousMinimap && previousMinimap.destroy()
wavesurfer.registerPlugin(minimap)
```

**Comment by Kalinkindev** [comment]
good, thanks, the only thing is it doesn’t delete the previous minimap, I’m looking for a problem why it adds a block, but doesn’t draw the waveform

![image](https://github.com/katspaugh/wavesurfer.js/assets/86912014/869496b4-ff32-4b78-8e5e-fd9b2c365a23)

code 
```
  wavesurfer.on("play", () => {
        const playerActive = document.querySelector(".audio--playActive");
        const playerActiveWafesurfer = playerActive.querySelector(".audio-webform");

        const minimap = Minimap.create({
          height: 30,
          container: playerActiveWafesurfer,
          waveColor: "#bebebe",
          progressColor: "#000000",
        });
        const previousMinimap = wavesurfer.getActivePlugins().find((plugin) => plugin instanceof Minimap);
        previousMinimap && previousMinimap.destroy();

        wavesurfer.registerPlugin(minimap);
        const timeDurationActive = playerActive.querySelector(".timeDuration");
        const timeCurrentActive = playerActive.querySelector(".timeCurrent");
        timeDurationActive.textContent = formatTime(wavesurfer.getDuration());

        wavesurfer.on("timeupdate", (currentTime) => {
          timeCurrentActive.textContent = formatTime(currentTime);
        });
      });
```

**Reply by katspaugh** [reply]
You might want to pass `overlayColor: 'transparent'` to the minimap options depending on what you want to achieve.

**Comment by Kalinkindev** [comment]
I want a logic like this https://wavesurfer.xyz/examples/?react-global-player.js

when playing audio
a minimap is created at the bottom of the page, when other audio is played, its minimap is created
```<div part="minimap" style="position: relative;"><div style="position: absolute; z-index: 2; left: 0px; top: 0px; bottom: 0px; transition: left 100ms ease-out 0s; pointer-events: none; background-color: rgba(100, 100, 100, 0.1); width: 138%;"></div></div>```
10 audio =1 waveform, not 10 audio = 10 waveform

in the screenshot above, a minimap is attached to each audio and when another audio is played, it is not deleted and an empty block is simply added, as if it was interrupted
what you sent works, but for some reason it doesn’t always draw the waveform
 ```
   const previousMinimap = wavesurfer.getActivePlugins().find((plugin) => plugin instanceof Minimap);
        previousMinimap && previousMinimap.destroy();
        wavesurfer.registerPlugin(minimap);
```
what you sent works, but for some reason it doesn’t always draw the waveform and creates its own minimap for each audio,
you probably need to remove minimap first

Sorry for my English, I'm writing through a translator

**Reply by Kalinkindev** [reply]
https://i.gyazo.com/b977cc1512bb17b8a75f0616ff0d1b98.png

it worked Waveform 


it worked, refreshed the page and stopped, I could send you a link to the page tomorrow if needed
